### ISSUE TITLE
nupkg packaging of System.IO.Compression is broken for net472

### ISSUE THREAD

**Issue by mgravell** [dotnet/corefx#31997]
In reference to [System.IO.Compresssion](https://www.nuget.org/packages/System.IO.Compression/), current version is 4.3.0

Isssue *only* seems to impact `net472`

Example (designed to test multiple frameworks)

```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFrameworks>netstandard2.0;net461;net47;net471;net472</TargetFrameworks>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="System.IO.Compression" Version="4.3.0" />    
  </ItemGroup>
</Project>
```

with

```c#
using System;
using System.IO.Compression;
namespace ClassLibrary1
{
    public static class Class1
    {
        public static void Bar()
        {
            ZipArchive z = null;
            GC.KeepAlive(z);
        }
    }
}
```

This compiles fine for all frameworks *except* `net472`. This fails to resolve the assembly correctly:

```
C:\Users\mgravell\source\repos\ConsoleApp14\ClassLibrary1>dotnet build -c Release -f net472
Microsoft (R) Build Engine version 15.7.179.6572 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 50.9 ms for C:\Users\mgravell\source\repos\ConsoleApp14\ClassLibrary1\ClassLibrary1.csproj.
Class1.cs(9,13): error CS0246: The type or namespace name 'ZipArchive' could not be found (are you missing a using directive or an assembly reference?) [C:\Users\mgravell\source\repos\ConsoleApp14\ClassLibrary1\ClassLibrary1.csproj]

Build FAILED.

Class1.cs(9,13): error CS0246: The type or namespace name 'ZipArchive' could not be found (are you missing a using directive or an assembly reference?) [C:\Users\mgravell\source\repos\ConsoleApp14\ClassLibrary1\ClassLibrary1.csproj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:00.91
```

**Comment by mgravell** [comment]
Observation: nothing in the existing nupkg targets net47* - it stops at `<group targetFramework=".NETFramework4.6" />` - so I guess this is something in the resolver itself that makes `net472`  pick a different path than `net47`/`net471`

**Comment by mgravell** [comment]
Closing; this seems to have already been fixed - the problem above happens when using 2.1.300, but is fixed when using 2.1.401

I'd still love to know what the problem was/is, though, as it may impact my consumers.
